### PR TITLE
[IMP] mail: introduce EmojiView model

### PR DESCRIPTION
**Pull request by Nemael**
Task-2365687

### CODE REVIEW

File: addons/mail/static/src/models/emoji_registry.js [hunk 1]
```diff
@@ -94,5 +94,8 @@ registerModel({
             ]),
             inverse: 'emojiRegistry',
         }),
+        emojiViews: many('EmojiView', {
+            inverse: 'emojiRegistry',
+        }),
```
Comment (alexkuhn) [inline addons/mail/static/src/models/emoji_registry.js:99]: can remove this (useless)

File: addons/mail/static/src/models/emoji_view.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+registerModel({
+    name: 'EmojiView',
+    identifyingFields: ['messaging'],
+    fields: {
+        emojiRegistry: one('EmojiRegistry', {
+            inverse: 'emojiViews',
+            readonly: true,
+            required: true,
+        }),
```
Comment (alexkuhn) [inline addons/mail/static/src/models/emoji_view.js:14]: same, can remove as it is useless

File: addons/mail/static/src/models/emoji_registry.js [hunk 3]
```diff
@@ -52,7 +52,7 @@ registerModel({
                 { unicode: "🌹", sources: [":sunflower"], description: ":sunflower" },
                 { unicode: "🔥", sources: [":fire"], description: ":fire" },
                 { unicode: "☀️", sources: [":sun"], description: ":sun" },
-                { unicode: "⛅️", sources: [":partly_sunny:"], description: ":partly_sunny:" },
+                { unicode: "⛅️", sources: [":partly_sunny"], description: ":partly_sunny" },
```
[the remark this answers is not in the document]
Comment (Nemael) [reply addons/mail/static/src/models/emoji_registry.js:55]: That was a discussion with Louis, I removed this to make it uniform with the syntax of the other emojis. I think it make sense to make all emojis more complex than faces have the same kind of syntax, either ":" in front, or ":" in front and back, like Discord

**Comment by alexkuhn** [comment]
rebase to resolve conflicts on `models/messaging.js` (if any)

**Comment by alexkuhn** [comment]
Likely there are indeed conflicts to resolve, because this PR that has just been merged made changes in `models/messaging.js`:
https://github.com/odoo/odoo/pull/92327
Make sure to resolve conflicts so that it takes correctly both changes into account on `models/messaging.js`.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+ rebase-ff

**Pull request closed without merging**